overlord/snapstate/backend/link.go

```diff
@@ -406,5 +406,9 @@ func (b Backend) UnlinkComponent(cpi snap.ContainerPlaceInfo, snapRev snap.Revis
 		}
 	}
 
+	// Try also to remove the <snap_rev>/ subdirectory, as this might be
+	// the only installed component. But simply ignore if not empty.
+	os.Remove(filepath.Dir(linkPath))
```

Comment:
historically just ignoring errors is what the code here does with os.Remove,  we could make the change but it would need to be a larger separate change with some further discussion